packages/cozy-stack-client/src/PermissionCollection.js

```diff
@@ -104,15 +104,23 @@ class PermissionCollection extends DocumentCollection {
     return { ...resp, data: resp.data.map(a => ({ _id: a.id, ...a })) }
   }
 
-  async createSharingLink(document) {
+  /**
+   * Create a share link
+   *
+   * @param {{_id, _type}} document - cozy document
+   * @param {object} options - options
+   * @param {string[]} options.verbs - explicit permissions to use
+   */
+  async createSharingLink(document, options = {}) {
+    const { verbs } = options
     const resp = await this.stackClient.fetchJSON(
       'POST',
       `/permissions?codes=email`,
       {
         data: {
           type: 'io.cozy.permissions',
           attributes: {
-            permissions: getPermissionsFor(document, true)
+            permissions: getPermissionsFor(document, verbs ? { verbs } : true)
```

Comment:
I don't think this works as intended, if we pass `verbs` we will call `getPermissionsFor (document, options)` bu the signature is `getPermissionsFor (document, publicLink, options)`

Reply:
Stupid me, should be better now, and I added tests